CMakeLists.txt

```diff
@@ -478,6 +478,66 @@ if (OPENSSL_FOUND AND CUDAQ_ENABLE_REST)
   add_subdirectory(tpls/Crow)
 endif()
 
+# AWS SDK for C++
```

Comment:
This should probably only be built if OPENSSL_FOUND and CUDAQ_ENABLE_REST, i.e. expand the if-block above to include this. Alternatively, it may make sense to migrate the build of the aws sdk into the cudaq/platform/default/rest/CMakeLists.txt, where it is actually needed.

Reply:
Partially addressed in 63d84bbbe956e72a2737b6cce1123ef301823002 - keeping it top-level, will consider moving into `rest` directory separately.